explaining-ratings/src/app.py

```diff
@@ -85,52 +94,69 @@ def handle_filter(q: Q, key: str, val: str):
         q.client.filters[key] = val
 
 
+async def update_theme(q: Q):
+    """
+    Update theme of app.
+    """
+    if q.args.theme_dark:
+        # Update theme from light to dark mode
+        q.page['meta'].theme = 'h2o-dark'
+    else:
+        # Update theme from dark to light mode
+        q.page['meta'].theme = 'light'
+
+    await q.page.save()
+
+
 @app('/')
 async def serve(q: Q):
     if not q.client.initialized:
         init(q)
         q.client.filters = {}
         q.client.initialized = True
 
-    args = expando_to_dict(q.args)
-    for arg in args:
-        if str(arg).startswith('filter_'):
-            handle_filter(q, str(arg).replace('filter_', ''), q.args[arg])
-
-    sidebar_items = q.page['sidebar'].items
-    if q.args.filter_categories:
-        # TODO: Remove after https://github.com/h2oai/wave/issues/150 gets resolved.
-        sidebar_items[2].dropdown.value = q.args.filter_categories
-    if q.args.filter_city:
-        # TODO: Remove after https://github.com/h2oai/wave/issues/150 gets resolved.
-        sidebar_items[3].dropdown.value = q.args.filter_city
-    if q.args.filter_country:
-        # TODO: Remove after https://github.com/h2oai/wave/issues/150 gets resolved.
-        sidebar_items[4].dropdown.value = q.args.filter_country
-    if q.args.filter_postalCode:
-        # TODO: Remove after https://github.com/h2oai/wave/issues/150 gets resolved.
-        sidebar_items[5].dropdown.value = q.args.filter_postalCode
-    if q.args.filter_province:
-        # TODO: Remove after https://github.com/h2oai/wave/issues/150 gets resolved.
-        sidebar_items[6].dropdown.value = q.args.filter_province
-    if q.args.filter_rating:
-        # TODO: Remove after https://github.com/h2oai/wave/issues/150 gets resolved.
-        sidebar_items[7].dropdown.value = q.args.filter_rating
-    if q.args.filter_userCity:
-        # TODO: Remove after https://github.com/h2oai/wave/issues/150 gets resolved.
-        sidebar_items[8].dropdown.value = q.args.filter_userCity
-    if q.args.filter_userProvince:
-        # TODO: Remove after https://github.com/h2oai/wave/issues/150 gets resolved.
-        sidebar_items[9].dropdown.value = q.args.filter_userProvince
-    if q.args.review:
-        q.client.review = q.args.review
-        # TODO: Remove after https://github.com/h2oai/wave/issues/150 gets resolved.
-        sidebar_items[0].dropdown.value = q.client.review
-        q.page['original'].image = plot_word_cloud(config.dataset[q.client.review], q)
-
-    if q.client.filters:
-        render_diff_word_cloud(q)
+    if q.args.theme_dark is not None and q.args.theme_dark != q.client.theme_dark:
+        await update_theme(q)
     else:
-        del q.page['diff']
```

Comment:
What's the reason for including the rest of the handlers in an `else` block? Can it be flattened (else removed)?

Reply:
if they are not in `else` block when we switch them it resubmits the filters as well. So the rating diff also getting refreshed

Reply:
But if the same filters are applied again, shouldn't it result in the very same diff? In which case it doesn't matter if they are rerun or not.

Reply:
@mturoci no it doesn't it creates different word clouds (as diff) for each time it get triggered with same filters